modules/nanopolish/1.0/nanopolish.smk

```diff
@@ -0,0 +1,266 @@
+
+# End of dependency checking section 
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["promethion"]`
+CFG = op.setup_module(
+    name = "nanopolish",
+    version = "1.0",
+    subdirectories = ["inputs", "meth_calls", "meth_freq","merged_nanopolish_calls","merged_nanopolish_freq","outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _promethion_input,
+    _install_nanopolish,
+    _promethion_input_chrs,
+    _nanopolish_output,
+    _nanopolish_all
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _promethion_input:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"],
+        fastq = CFG["inputs"]["sample_fastq"],
+        index = CFG["inputs"]["sample_index"],
+        fai = CFG["inputs"]["sample_fai"],
+        gzi = CFG["inputs"]["sample_gzi"],
+        readdb = CFG["inputs"]["sample_readdb"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai",
+        fastq = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq",
+        index = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index",
+        fai = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.fai",
+        gzi = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.gzi",
+        readdb = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.readdb",
```

Comment:
We don't have these .index.fai/gzi/readdb files symlinked into GAMBL. Are they generated by default for each fastq file by the GSC?

Reply:
I see that you generated them within the data directories for the BLGSP promethION data. Could you add a rule or rules to generate these files in the `00-inputs` directory alongside the fastq files after you've symlinked them in?

Reply:
These files are actually generated by the GSC and are in the same directory as the fastqs. I don't know if it makes sense to just symlink them into gambl under the fastq directory?

Reply:
Got it! I can ask Haya to update the GAMBL script that symlinks the PromethION data to include these files.